### PR TITLE
Replace kv/rest's router with a httprouter.

### PR DESCRIPTION
**Pull request by BramGruneir**
https://github.com/julienschmidt/httprouter

This new router is faster, smaller and allows for parsing of parameters using a pattern in the path as opposed to having to parse it ourselves.  

I plan to clean up other instances as well.

### CODE REVIEW

File: kv/rest.go [hunk 1]
```diff
@@ -122,39 +124,14 @@ func writeJSON(w http.ResponseWriter, statusCode int, v interface{}) {
...
-	result, err := url.QueryUnescape(strings.TrimPrefix(path, apiPrefix))
-	if err == nil {
-		k := proto.Key(result)
-		if len(k) == 0 {
-			return nil, errors.New("empty key not allowed")
-		}
-		return k, nil
-	}
-	return nil, err
+// handleEmptyKey returns a specific error instead of a generic 404 is no key
```
Comment (spencerkimball) [inline kv/rest.go]: s/404 is/404 if/

**Comment by spencerkimball** [comment]
LGTM. Seems like a huge improvement.

File: kv/rest.go (hunk 1 above)
Comment (BramGruneir) [inline kv/rest.go]: Fixed.

**Comment by tamird** [comment]
Nice. @BramGruneir: your git config looks broken, though :(

**Comment by mrtracy** [comment]
LGTM

**Pull request merged**